### PR TITLE
Fmendez/auto promote

### PR DESCRIPTION
**Pull request by fmendez884**

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -98,7 +98,7 @@ type SquareActor = Actor & Coordinate;
 const modelConfigs: { [id: string]: { [id: string]: ModelConfig } } = {
 	black: {
 		rook: { rotation: Quaternion.Identity() },
-		knight: { rotation: Quaternion.Identity() },
+		knight: { rotation: Quaternion.Identity() },
```
Comment (tyboogie) [inline src/app.ts:101]: [Nit] Remove whitespace

File: src/app.ts [hunk 2]
```diff
@@ -159,14 +160,15 @@ export default class ChessGame {
...
+			this.createResetButton()
 		]);
 
 		// Hook up event handlers.
 		// Do this after all actors are loaded because the event handlers themselves reference other actors in the
 		// scene. It simplifies handler code if we can assume that the actors are loaded.
 		this.addEventHandlers();
 	}
-
+
```
Comment (tyboogie) [inline src/app.ts:171]: [Nit] Remove whitespace

File: src/app.ts [hunk 3]
```diff
@@ -351,9 +397,17 @@ export default class ChessGame {
 			actor.onGrab('begin', (user) => this.onDragBegin(user.id, actor));
 			actor.onGrab('end', (user) => this.onDragEnd(user.id, actor));
 			actor.grabbable = true;
-		});
+		});this.addResetButtonEventHandlers();
```
Comment (tyboogie) [inline src/app.ts:400]: [Nit] Call this function on a newline.

**Review by tyboogie — COMMENTED** [review]
Some nitpicks but LGTM.  I'll push to @eanders-ms to see what he thinks.

File: src/app.ts [hunk 4]
```diff
@@ -390,7 +452,16 @@ export default class ChessGame {
 					.filter(item => item.file === dropSquare.file && item.rank === dropSquare.rank).shift();
 				if (destSquare) {
 					// Move the piece.
-					this.game.move(move.src, destSquare);
+					if (move.src.piece.type === 'pawn' && (destSquare.rank === 8 || destSquare.rank === 1)) {
+						//Auto promote pawn to queen
+						this.game.move(move.src,destSquare, "Q");
+						
+						const newActor = this.promoteChessPiece(userId, actor, destSquare);
+						await Promise.resolve(newActor);
```
Comment (eanders-ms) [inline src/app.ts:460]: Why is the call to Promise.resolve necessary? Would this work:
```ts
actor = this.promoteChessPiece(userId, actor, destSquare);
```

Comment (fmendez884) [reply src/app.ts:460]: Yes that does work. I'm pushing the change now

File: src/app.ts [hunk 5]
```diff
@@ -205,7 +207,31 @@ export default class ChessGame {
 			.then(value => this.preloads['check-marker'] = value));
 		await Promise.all(preloads);
 	}
+	
+	private async loadActorsAndEvents() {
```
Comment (eanders-ms) [inline src/app.ts:211]: There's a lot of code duplication here with the contents of the `started` method. I think you may be able to just call `this.resetGame()` in started now. Something like:
```ts
private started = async () => {
    this.resetGame();
}
```

Comment (fmendez884) [reply src/app.ts:195]: Just pushed the changes. I also moved the resetGame() function closer to the top, near started()

File: src/app.ts [hunk 6]
```diff
@@ -132,6 +132,7 @@ export default class ChessGame {
 	private checkMarker: Actor;
 	private assets: AssetContainer;
 	private preloads: { [id: string]: Asset[] } = {};
+	private resetButton: Actor;
```
Comment (eanders-ms) [inline src/app.ts:135]: Nit: Would be nice if `resetButton` were grouped with the other actors above.

File: src/app.ts [hunk 7]
```diff
@@ -206,6 +186,36 @@ export default class ChessGame {
...
+			this.createJoinButtons(),
+			this.createResetButton()
+		]);
+
+		// Hook up event handlers.
+		// Do this after all actors are loaded because the event handlers themselves reference other actors in the
+		// scene. It simplifies handler code if we can assume that the actors are loaded.
+		this.addEventHandlers();
+	}
+
```
Comment (eanders-ms) [inline src/app.ts:218]: Nit: There's a lot of trailing whitespace on blank lines in the file. Suggest letting vscode format the file (Alt+Shift+F hotkey).

**Review by eanders-ms — APPROVED** [review]
Two small nits, but LGTM!

**Pull request merged**